### ISSUE TITLE
dotnet test log results issue with a filename that includes a space - Can't escape the space

### ISSUE THREAD

**Issue by pmcgrath** [dotnet/cli#7650]
I know spaces in file paths are a terrible idea, but I need to deal with this due to the CI system I am using which clones repos into a path that includes a directory with a space in its name
I also want all of the test results for multiple projects in a single output directory

I have used a simplistic scenario to illustrate

What I would like to know is if i can escape a space in the path ?

## Steps to reproduce
Using powershell on windows
```
cd /tmp
mkdir test-issue
cd ./test-issue
ls
dotnet new xunit
dotnet test
dotnet test --logger 'trx;logfilename=res.trx'

# Now lets try to use a log file name with a space - Failure !
dotnet test --logger 'trx;logfilename=test res.trx'
> Microsoft (R) Test Execution Command Line Tool Version 15.3.0-preview-20170628-02
> Copyright (c) Microsoft Corporation.  All rights reserved.

> The test source file "C:\tmp\test-issue\res.trx" provided was not found.
> The test source file "C:\tmp\test-issue\res.trx" provided was not found.

# Tried the following escaping all failed
dotnet test --logger 'trx;logfilename="test res.trx"'
dotnet test --logger "trx;logfilename='test res.trx'"
```

## Expected  behavior
Would like to be able to escape the logfilename path

## Actual behavior
Can't escape the logfilename path

## Environment data
.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\utilities\dotnet-core-sdks\sdk\2.0.0\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

**Comment by peterhuene** [comment]
Hi @pmcgrath.  Thanks for reporting this issue!

With the latest build out of the `master` branch, **I'm unable** to reproduce the problem.

With PowerShell (as it supports both `'` and `"` for arguments):
```ps1
dotnet test --logger 'trx;logfilename=test res.trx'
...
Results File: ...\test res.trx
```

With cmd.exe (supports only `"` for arguments):
```cmd
dotnet test --logger "trx;logfilename=test res.trx"
...
Results File: ...\test res.trx
```

**I am able** to reproduce the issue with 2.0.0, however.

I'm looking into tracking down what might have changed to fix this issue.  From the error you're seeing, I suspect the `/p` argument to msbuild wasn't being properly quoted in 2.0.0, resulting in "res.trx" being interpreted as a separate argument to vstest.

**Comment by peterhuene** [comment]
I've tracked down the fix which appears to have been made to the `VSTest` task: https://github.com/Microsoft/vstest/commit/a2c31a98fbce35a66bef8932665d4d79ed082bb0#diff-13cf5bf02226df4726f8a932f681df8

**Comment by pmcgrath** [comment]
@peterhuene Thanks, so 2.1.0 release will include this fix ? 
I'm getting the same behavior on 2.0.3 as 2.0.0

**Comment by peterhuene** [comment]
@pmcgrath good question :)

@livarcocc is there an easy way to tell if a commit from a repo like microsoft/vstest is going to be in a particular release of the sdk?

**Comment by livarcocc** [comment]
Unfortunately, no. I would just test 2.1.2 (coming out of release/2.1) and see if it is in there. That's our current release tip. If it is not there, then yeah, it is in master only.

**Comment by peterhuene** [comment]
@pmcgrath It does appear the fix is in the `release/2.1` branch.

@livarcocc Good to close?

**Comment by livarcocc** [comment]
Thanks @peterhuene for following through on this issue.
